reference/src/layout/unions.md

```diff
@@ -6,17 +6,109 @@ not to change until an RFC ratifies them.
 
 [#13]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/13
 
-The only degree of freedom the compiler has when computing the layout of a union
```

Comment:
I like this "the only degree of freedom" formulation, could you keep it?

Reply:
However, the old sentence is not entirely correct.

I would write something like

"A union consists of several *variants*, one for each field. These variants are laid out "on top" of each other, so they must all have the same size. The only degree of freedom the compiler has when computing the layout of a union is to pick where in its variant each field is situated, i.e., the compiler picks the gap (often called padding) before and after each field."

Reply:
Does the compiler only have one degree of freedom? I think it can, at least, choose: 

* offset of fields,
* amount of trailing padding (e.g. we don't guarantee an upper bound for this),

within certain constraints.

Reply:
Well maybe call it "degrees of freedom" then. What you are saying is all already contained in my proposed wording.